File: app/models/qdm/specimen_definition.rb
module QDM
  # app/models/qdm/specimen_definition.rb
  class SpecimenDefinition < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :typeCollected
    field :patientPreparation
    field :timeAspect
    field :collection
    field :typeTested
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/SpecimenDefinition.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const SpecimenDefinitionSchema = DataElementSchema({
  identifier: Any,
  typeCollected: Any,
  patientPreparation: Any,
  timeAspect: Any,
  collection: Any,
  typeTested: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'SpecimenDefinition' },

});

module.exports.SpecimenDefinitionSchema = SpecimenDefinitionSchema;
module.exports.SpecimenDefinition = mongoose.model('SpecimenDefinition', SpecimenDefinitionSchema);
